core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -504,16 +510,23 @@ private void registerBuiltinSorts() {
         namedWriteableRegistry.register(SortBuilder.class, FieldSortBuilder.NAME, FieldSortBuilder::new);
     }
 
-    private void registerBuiltinFunctionScoreParsers() {
-        registerFunctionScoreParser(new ScriptScoreFunctionParser());
-        registerFunctionScoreParser(new GaussDecayFunctionParser());
-        registerFunctionScoreParser(new LinearDecayFunctionParser());
-        registerFunctionScoreParser(new ExponentialDecayFunctionParser());
-        registerFunctionScoreParser(new RandomScoreFunctionParser());
-        registerFunctionScoreParser(new FieldValueFactorFunctionParser());
+    private void registerBuiltinScoreFunctionParsers() {
+        registerScoreFunction(ScriptScoreFunctionBuilder.FIELD, ScriptScoreFunctionBuilder::fromXContent, ScriptScoreFunctionBuilder.NAME,
```

Comment:
It felt like that was what @cbuescher was arguing for - that we don't encode the convention for writable name in the API.